### PR TITLE
cmake,deb: install system units using cmake 

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-systemd-deb-env/4865831b913a7a413055237d3acf3623c421a884/

**Comment by tchaikov** [comment]
retest this please.

File: systemd/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,26 @@
 set(CMAKE_INSTALL_SYSTEMD_SERVICEDIR "${CMAKE_INSTALL_LIBEXECDIR}/systemd/system"
   CACHE PATH "Location for systemd service files")
+set(CEPH_SYSTEMD_ENV_DIR "/etc/sysconf"
+  CACHE PATH "Location for systemd service environmental variable settings files")
```
Comment (smithfarm) [inline systemd/CMakeLists.txt]: with this change, the RPM spec file will need to run cmake with `-DCEPH_SYSTEMD_ENV_DIR=/etc/sysconfig`?

**Comment by smithfarm** [comment]
@tchaikov Looks good to me, as long as it doesn't break RPM :-)

File: systemd/CMakeLists.txt (hunk 1 above)
Comment (tchaikov) [reply systemd/CMakeLists.txt]: no need to do so. as the default value is "/etc/sysconf".

**Comment by tchaikov** [comment]
the rpm build passed. see https://github.com/ceph/ceph/pull/20618#issuecomment-369082663

**Comment by tchaikov** [comment]
@smithfarm sorry, i merged before i noticed that you requested more reviewers. please @  me if anything does not look right.

**Comment by tchaikov** [comment]
will post a fix for s/sysconf/sysconfig/

**Comment by tchaikov** [comment]
#20631

**Pull request merged**